### ISSUE TITLE
call for geo index hackers

### ISSUE THREAD

**Issue by biolee** [pingcap/tidb#6347]
> We may reconsider the feature later. But it will be no early than the end of this year.

pingcap do not have plan for geo index https://github.com/pingcap/tidb/issues/673#issuecomment-251132927,  I am interested in adding this feature to tidb, so Is there anyone else interested or working on this?

**Comment by XuHuaiyu** [comment]
@biolee  Thank you for your kind help!

**Comment by coocood** [comment]
@biolee 
Thank you for willing to help!
But this feature may take hundreds of hours, are you prepared?

**Comment by morgo** [comment]
This is currently documented under [MySQL Compatibility](https://www.pingcap.com/docs/sql/mysql-compatibility/#unsupported-features).

**Comment by scsldb** [comment]
We need to investigate the user scenario in depth, and then make a decision after the research is completed.

**Comment by awsl-dbq** [comment]
Hi @unship , for 4 years later, do you still interested in it?

**Comment by xxscloud5722** [comment]
It’s 2024. Does gis have any plans?

**Comment by dveeden** [comment]
> It’s 2024. Does gis have any plans?

Yes there are plans. But this will take many steps and I can't commit to a timeline yet.

**Comment by dveeden** [comment]
My plan is to this in multiple steps

1. Support geospatial types and srid constraints in the parser
2. Support geospatial types and conversion functions for wkt to wkb and back
3. Support more functions
4. Support spatial indexes
